src/wallet/crypter.cpp

```diff
@@ -135,12 +136,31 @@ static bool DecryptKey(const CKeyingMaterial& vMasterKey, const std::vector<unsi
     return key.VerifyPubKey(vchPubKey);
 }
 
+static bool DecryptSpendingKey(const CKeyingMaterial& vMasterKey,
+                               const std::vector<unsigned char>& vchCryptedSecret,
+                               const libzcash::PaymentAddress& address,
+                               libzcash::SpendingKey& sk)
+{
+    CKeyingMaterial vchSecret;
+    if(!DecryptSecret(vMasterKey, vchCryptedSecret, address.GetHash(), vchSecret))
+        return false;
+
+    if (vchSecret.size() != libzcash::SerializedSpendingKeySize)
+        return false;
+
+    // TODO does this undo the benefits of using CKeyingMaterial?
```

Comment:
`CDataStream` internally copies the input data into a `CSerializeData`, which is defined as:

```
// Byte-vector that clears its contents before deletion.
typedef std::vector<char, zero_after_free_allocator<char> > CSerializeData;
```

We could instead alter it to use `secure_allocator`, which locks its contents from being paged out of memory and clears its contents before deletion. But I expect this would cause performance issues elsewhere in the code. So instead I'm going to subclass (or probably just copy) `CDataStream` to `CSecureDataStream` and use a `secure_allocator` there.